instrumentation/instaredigo/instaredigo.go

```diff
@@ -0,0 +1,175 @@
+)
+
+type instaRedigoConn struct {
+	redis.Conn
+	sensor   *instana.Sensor
+	address  string
+	prevSpan *prevSpan
+}
+
+type prevSpan struct {
+	span      ot.Span
+	batchCmds []string
+}
+
+//Dial connects to the Redis server at the given network and address using the specified options along with instrumentation code.
+func Dial(sensor *instana.Sensor, network, address string, options ...redis.DialOption) (redis.Conn, error) {
+	conn, err := redis.Dial(network, address, options...)
+	if strings.HasPrefix(address, ":") {
+		address = "localhost" + address
+	}
+	return &instaRedigoConn{Conn: conn, sensor: sensor, address: address, prevSpan: nil}, err
+}
+
+//DialURL wraps DialURLContext using context.Background along with the instrumentation code.
+func DialURL(sensor *instana.Sensor, rawurl string, options ...redis.DialOption) (redis.Conn, error) {
+	conn, err := redis.DialURL(rawurl, options...)
+	return &instaRedigoConn{conn, sensor, rawurl, nil}, err
+}
+
+//DialURLContext connects to a Redis server at the given URL using the Redis URI scheme along with the instrumentation code.
+func DialURLContext(sensor *instana.Sensor, ctx context.Context, rawurl string, options ...redis.DialOption) (redis.Conn, error) {
+	conn, err := redis.DialURLContext(ctx, rawurl, options...)
+	return &instaRedigoConn{conn, sensor, rawurl, nil}, err
+}
+
+//NewConn returns a new Redigo connection for the given net connection along with the instrumentation code.
+func NewConn(sensor *instana.Sensor, netConn net.Conn, readTimeout, writeTimeout time.Duration) redis.Conn {
+    addr := netConn.LocalAddr().String()
+	conn := redis.NewConn(netConn, readTimeout, writeTimeout)
+	return &instaRedigoConn{conn, sensor, addr, nil}
+}
+
+/// Do sends a command to the server and returns the received reply and collect the instrumentation details. Inorder to capture the correlated span information, create a context from the parent span and pass it as an argument along with the other arguments. The Do API will retrieve the span information from the context and record the correlated span information.
+func (c *instaRedigoConn) Do(commandName string, args ...interface{}) (reply interface{}, err error) {
+	var cmdArgs []interface{}
+	ctx := context.Background()
+    //Separating the passed context and the arguments
+	for _, a := range args {
+		if _, ok := a.(context.Context); ok {
+			ctx = a.(context.Context)
+		} else {
+			cmdArgs = append(cmdArgs, a)
+		}
+	}
+	if c.prevSpan != nil {
+		if strings.ToUpper(commandName) == "EXEC" || strings.ToUpper(commandName) == "DISCARD" {
+			reply, err = c.Conn.Do(commandName, cmdArgs...)
+			if err != nil {
+			    c.prevSpan.span.SetTag("redis.error", err.Error())
+				c.prevSpan.span.LogFields(otlog.Object("error", err.Error()))
+			}
+			c.prevSpan.span.Finish()
+			c.prevSpan = nil
+		} else {
+			c.prevSpan.batchCmds = append(c.prevSpan.batchCmds, commandName)
+			c.prevSpan.span.SetTag("redis.subCommands", c.prevSpan.batchCmds)
```

Comment:
Also not a big deal, but mostly a personal preference: You could set the tag only once in the `if` part, just before setting `c.prevSpan` to nil :)